src/states/proving_node.rs

```diff
@@ -248,11 +329,14 @@ impl Base for ProvingNode {
     }
 
     fn handle_connect_failure(&mut self, pub_id: PublicId, outbox: &mut EventBox) -> Transition {
+        debug!("{} received connection failure from {:?}", self, pub_id);
+        let _ = self.resource_proofing_status.remove(&pub_id);
         RelocatedNotEstablished::handle_connect_failure(self, pub_id, outbox)
     }
 
     fn handle_lost_peer(&mut self, pub_id: PublicId, outbox: &mut EventBox) -> Transition {
         debug!("{} Received LostPeer - {}", self, pub_id);
+        let _ = self.resource_proofing_status.remove(&pub_id);
```

Comment:
Why is that need to change?

Reply:
If it's not received, the disconnected peer will be left over in the `resource_proofing_status` map, which will cause the `connection_info_req` to be sent out forever and exceeding the polling max_limitation

Reply:
Why did that change with this PR? why did it not happen before as well?

Reply:
The `resource_proofing_status` map is introduced by this PR only. 

Previously, a disconnected connection is will have only one `disconnected` msg to be sent to peer.
But now, with the new `resending` attempt, if the entry doesn't get cleaned up, we will send out ConnectionInfo request each time reached timeout.

Reply:
Thanks, yes I got confused there.
still, should it not be set to false instead?

> But now, with the new resending attempt, if the entry doesn't get cleaned up, we will send out ConnectionInfo request each time reached timeout.

Won't it eventually stop when reaching the trying to prove themselves timeout?

Reply:
>Won't it eventually stop when reaching the trying to prove themselves timeout
No. As long as the node is in the proving_node state and having the map which contains the entry, the connection_info_req will be sent out each time a timeout is reached.

Reply:
Yes, that is what it is suppose to do, but does it not move to Terminated state?

Reply:
> does it not move to Terminated state

No. there is no timeout on proving_node itself.
There is only timeout on the elders side which they just reject and disconnect.
But because the proving_node itself keeps sending out msg, the `poll` loop won't break and exceeds the max